File: frontend/lib/domain/model/workout/workout_summary.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'workout_summary.freezed.dart';

@freezed
class WorkoutSummary with _$WorkoutSummary {
  const factory WorkoutSummary({
    required int id,
    required String name,
    required int totalDuration,
    required String category,
    required DateTime createdAt,
    required DateTime updatedAt,
  }) = _WorkoutSummary;
}

File: backend/app/api/entities/workout_summary.rb
module Entities
  class WorkoutSummary < Grape::Entity
    expose :id
    expose :name
    expose :total_duration
    expose :category, documentation: { desc: 'カテゴリ名' } do |summary|
      summary.category.name
    end
    expose :created_at
    expose :updated_at
  end
end
